linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1971,8 +1973,42 @@ private void closeInner() {
       final Function2<TSource, TInner, TResult> resultSelector,
       boolean generateNullsOnLeft,
       boolean generateNullsOnRight) {
+    if (generateNullsOnLeft) {
+      throw new UnsupportedOperationException(
+          "not implemented, mergeJoin with generateNullsOnLeft");
+    }
+    if (generateNullsOnRight) {
+      throw new UnsupportedOperationException(
+          "not implemented, mergeJoin with generateNullsOnRight");
+    }
+    return mergeJoin(outer, inner, outerKeySelector, innerKeySelector, null, resultSelector,
+        JoinType.INNER, null);
+  }
+
+  public static boolean isMergeJoinSupported(JoinType joinType) {
```

Comment:
Can we add a note that this method is subject to change or remove without notice?

Reply:
Sure, I can add that; but can you please elaborate why do think it is necessary to mention it?

Reply:
Once all the join type are supported, there is no need for the method.

Reply:
That is an optimistic point of view :) 
Ok, I'll add the note.

Reply:
:) Indeed, it is up to you.

Reply:
done